Code/GraphMol/RascalMCES/RascalResult.h

```diff
@@ -0,0 +1,176 @@
+
+  RascalResult(const RascalResult &other);
+
+  RascalResult(RascalResult &&other) = default;
+
+  ~RascalResult() = default;
+
+  RascalResult &operator=(const RascalResult &other);
+
+  RascalResult &operator=(RascalResult &&other) = default;
+
+  // Cut the result down to the single largest fragment.  This is
+  // irrecoverably destructive.
+  void largestFragOnly();
+  void largestFragsOnly(unsigned int numFrags = 2);
+  void trimSmallFrags(unsigned int minFragSize = 3);
+
+  std::shared_ptr<RDKit::ROMol> getMol1() const { return d_mol1; };
+
+  std::shared_ptr<RDKit::ROMol> getMol2() const { return d_mol2; };
+
+  std::vector<std::pair<int, int>> getBondMatches() const {
+    return d_bondMatches;
+  }
+
+  std::vector<std::pair<int, int>> getAtomMatches() const {
+    return d_atomMatches;
+  }
+
+  int getNumFrags() const;
+
+  int getRingNonRingBondScore() const;
+
+  int getAtomMatchScore() const;
+
+  int getMaxDeltaAtomAtomDist() const;
+
+  // returns number of atoms for largest fragment.
+  int getLargestFragSize() const;
+
+  std::string getSmarts() const;
+  const std::shared_ptr<ROMol> getMcesMol() const;
+  bool getTimedOut() const { return d_timedOut; };
+
+  double getTier1Sim() const { return d_tier1Sim; }
+  double getTier2Sim() const { return d_tier2Sim; }
+  double getSimilarity() const;
+
+ private:
+  std::shared_ptr<ROMol> d_mol1;
+  std::shared_ptr<ROMol> d_mol2;
+  mutable std::shared_ptr<ROMol> d_mcesMol;
+  std::vector<std::pair<int, int>> d_bondMatches;
+  std::vector<std::pair<int, int>> d_atomMatches;
+
+  mutable std::string d_smarts;
+  bool d_timedOut{false};
+  double d_tier1Sim;
+  double d_tier2Sim;
+  bool d_ringMatchesRingOnly{false};
+  int d_maxFragSep{-1};
+
+  // These are used for sorting the results.
+  mutable int d_numFrags{-1};
+  mutable int d_ringNonRingBondScore{-1};
+  mutable int d_atomMatchScore{-1};
+  mutable int d_maxDeltaAtomAtomDist{-1};
+  mutable int d_largestFragSize{-1};
+
+  // Assuming the frags are all part of the original MCES, just cut it
+  // down to what's in the frags.
+  void rebuildFromFrags(const std::vector<boost::shared_ptr<ROMol>> &frags);
+
+  std::string createSmartsString() const;
+
+  void matchCliqueAtoms(const std::vector<std::vector<int>> &mol1_adj_matrix);
+
+  // If the clique involves a fragment that is more than d_maxFragSep from
+  // any other frag in either molecule, discard the smaller frag.
+  void applyMaxFragSep();
+
+  // Make the fragments for either mol1 or mol2.  If molNum is not 1 or 2,
+  // returns nullptr.
+  RDKit::ROMol *makeMolFrags(int molNum) const;
+
+  int calcRingNonRingScore() const;
+
+  int calcAtomMatchScore() const;
+
+  int calcLargestFragSize() const;
+
+  // If there are multiple fragments, can be helpful as a tie-breaker.  It's the
+  // maximum difference between through-bond distances between matching atoms in
+  // the 2 molecules.
+  int calcMaxDeltaAtomAtomDistScore() const;
+};
+
```

Comment:
How many of the functions defined in the rest of this header actually need to be in the public API?
Couldn't a lot of them be defined in the `RascalDetails` header?

Reply:
A further thought - cleanSmarts isn't perfect, but it does make the SMARTS strings much more legible, IMO.  I did wonder if it might be usefully exposed so fMCS could use it too.  But then I couldn't think where it would go, so I left it.